src/common/network/NetworkUtils.cpp

```diff
@@ -215,15 +215,32 @@ std::string NetworkUtils::intToIPv4(IPv4 ip) {
     return buf;
 }
 
-StatusOr<HostAddr> NetworkUtils::toHostAddr(folly::StringPiece ip, int32_t port) {
+StatusOr<std::vector<HostAddr>> NetworkUtils::toHostAddr(folly::StringPiece ip, int32_t port) {
```

Comment:
Given that now we don't have IPv6 support in mind, semanticly, we shall not return multiple addresses for a single hostname. So we could check on the list returned from `gethostbyname` to only yield the one that meets our criteria.

Reply:
This has nothing to do with ipv6, the host was likely to return multiple ip addresses.

Reply:
Multiple IP for single hostname? I have no idea.

Reply:
Because gethostbyname/getaddrinfo may take dns query, it is very common to return multiple ip addresses. Even if it is a local host, a host corresponding to multiple IP addresses is more convenient for operation and maintenance.

Reply:
Fair enough